### PR TITLE
DOCK-2406: Remove failing request on tool page

### PR DESCRIPTION
**Pull request by y-ng**
**Description**
When navigating to a public legacy tool page with browser tools open, there would be a workflow request that returns a 404 (as well as a similar request that succeeds so the page still shows properly). This PR prevents the failing request from being made on the tool page.

**Review Instructions**
With browser tools open, navigate to a public tool and check that there is no 404 in the network tab.

**Issue**
[DOCK-2406](https://ucsc-cgl.atlassian.net/browse/DOCK-2406)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone. 

[DOCK-2406]: https://ucsc-cgl.atlassian.net/browse/DOCK-2406?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/workflow/workflow.component.ts [hunk 1]
```diff
@@ -390,39 +390,41 @@ export class WorkflowComponent extends Entry<WorkflowVersion> implements AfterVi
...
-            // Retrieve the workflow path from the URL
-            const splitPath = this.resourcePath.split('/');
-            const workflowPath = splitPath.slice(2, 5);
-            const pathSuffix = workflowPath.join('/');
-
-            // Create suggested paths
-            this.gitlabPath += pathSuffix;
-            this.githubPath += pathSuffix;
-            this.bitbucketPath += pathSuffix;
+    if (!url.includes('containers') && !url.includes('tools')) {
```
Comment (y-ng) [inline src/app/workflow/workflow.component.ts:393]: I added this line so that the request isn't made on a legacy tool page. The integration and smoke tests pass but I wonder if this would prevent the request from being made on any pages it's actually needed.

**Review by coverbeck — COMMENTED** [review]
I think the fix may need to be in container.component.html and/or container.component.ts.

In [container.component.html](https://github.com/dockstore/dockstore-ui2/blob/541d618c19aaf7063f5740071952dc0d61e637fd/src/app/container/container.component.html#L446), the intent looks to be that the workflow.component.ts is only instantiated if it's an app tool, but it's getting invoked anyway.

I think the problem is that `tool` is initially undefined, so the template/component is invoked, which causes the the workflow component to get instantiated. Then tool gets assigned a legacy tool, the workflow component goes away, but it already made the Ajax request.

TLDR; this component shouldn't even exist for a legacy tool.

File: src/app/container/container.component.html [hunk 2]
```diff
@@ -13,7 +13,7 @@
   ~    See the License for the specific language governing permissions and
   ~    limitations under the License.
   -->
-<div *ngIf="tool; else workflow">
+<div *ngIf="tool || !displayAppTool; else workflow">
```
Comment (coverbeck) [inline src/app/container/container.component.html:16]: The template and property names were like this already, but this code makes it seem like there are 3 entity types to handle: Tool, AppTool, and Workflow. But this component handles just two types, Tool and AppTool, even though AppTool extends Workflow.

Suggest: change the template id to `appTool`, i.e., `<div *ngIf="tool || !displayAppTool; else appTool">`

**Review by coverbeck — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]
Think I accidentally dismissed a passing review when re-requesting

**Review by coverbeck — APPROVED** [review]

**Review by hyunnaye — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]
